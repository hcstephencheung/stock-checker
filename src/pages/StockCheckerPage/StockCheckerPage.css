.stockCheckerPage {
    padding: 24px;
}