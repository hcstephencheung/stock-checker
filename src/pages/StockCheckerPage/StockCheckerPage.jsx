import React, { useState } from "react";
import { useFetchCompanyDescription, useFetchStockSymbols, useFetchPriceFromSymbol } from "../../providers/StockListProvider";
import Price from "../../components/Price/Price";
import ContentBlock from "../../components/ContentBlock/ContentBlock";
import StockDropdown from "../../components/StockDropdown/StockDropdown";
import * as styles from "./StockCheckerPage.css";

const StockCheckerPage = () => {
    const [ symbol, setSymbol ] = useState("AAPL");
    const { price, priceLoading } = useFetchPriceFromSymbol(symbol);
    const { description, descriptionLoading } = useFetchCompanyDescription(symbol);
    const { symbols, symbolsLoading } = useFetchStockSymbols();

    const handleStockChange = e => setSymbol(e.target.value);

    return (
        <section className={styles.stockCheckerPage}>
            <StockDropdown isLoading={symbolsLoading} symbols={symbols} selectedSymbol={symbol} handleChange={handleStockChange} />
            <Price isLoading={priceLoading} title={"Current Stock Price"} price={price} currency={"USD"} />
            <ContentBlock isLoading={descriptionLoading} content={description} title={"Description"} />
        </section>
    )
}

export default StockCheckerPage;