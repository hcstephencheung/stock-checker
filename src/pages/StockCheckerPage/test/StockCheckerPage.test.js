import React from 'react';
import { shallow } from "enzyme";
import StockCheckerPage from "../StockCheckerPage";

describe("StockCheckerPage", () => {
    it("should render with skeletons", () => {
        const wrapper = shallow(<StockCheckerPage />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});

