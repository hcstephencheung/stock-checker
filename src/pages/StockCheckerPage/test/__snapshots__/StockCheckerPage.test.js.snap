// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockCheckerPage should render with skeletons 1`] = `
"<section className={[undefined]}>
  <StockDropdown isLoading={true} symbols={{...}} selectedSymbol=\\"AAPL\\" handleChange={[Function: handleStockChange]} />
  <Price isLoading={true} title=\\"Current Stock Price\\" price={{...}} currency=\\"USD\\" />
  <ContentBlock isLoading={true} content={{...}} title=\\"Description\\" />
</section>"
`;
